autogluon/task/text_classification/dataset.py

```diff
@@ -116,31 +120,168 @@ def get_dataset(self, segment='train'):
+
+        Returns
+        -------
+        tuple of str, Dataset : the segment name, and the dataset.
+        """
+        return 'dev', self.get_dataset(segment='dev')
+
+    def dataset_test(self):
+        """Get the test segment of the dataset for the task.
+
+        Returns
+        -------
+        tuple of str, Dataset : the segment name, and the dataset.
+        """
+        return 'test', self.get_dataset(segment='test')
+
+
+class CustomTSVClassificationTask(AbstractCustomTask):
+    """
+
+    Parameters
+    ----------
+    filepath: str, path of the file
+        Any valid string path is acceptable.
+    sep : str
+        Delimiter to use.
+    header : int, list of int, default 'infer'
+        Row number(s) to use as the column names, and the start of the
+        data.
+    names : array-like, optional
+        List of column names to use.
+    usecols : list-like or callable, optional
+        Return a subset of the columns.
+    For other keyword arguments, please see
+        <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    """
+    def __init__(self, *args, **kwargs):
+        self._read(**kwargs)
+        self.is_pair = False
+        self._label_column_id = self._index_label_column(**kwargs)
+        self.class_labels = list(set([sample[self._label_column_id] for sample in self.dataset]))
+        self.metric = Accuracy()
+        super(CustomTSVClassificationTask, self).__init__(self.class_labels, self.metric, self.is_pair)
+
+    def get_dataset(self, segment='train'):
+        """Get the corresponding dataset for the task.
+
+        Parameters
+        ----------
+        segment : str, default 'train'
+            Dataset segments.
+
+        Returns
+        -------
+        The dataset of target segment.
+        """
+        if segment == 'train':
+            return SampledDataset(self.dataset, self.train_sampler)
+        elif segment == 'dev':
+            return SampledDataset(self.dataset, self.dev_sampler)
+        else:
+            raise NotImplementedError
+
+    def _read(self, **kwargs):
+        path = kwargs.get('filepath', None)
+        kwargs['filepath_or_buffer'] = path
+        kwargs.pop("filepath")
+        low_memory = kwargs.get('low_memory', None)
+        if low_memory is None:
+            kwargs['low_memory'] = False
+        dataset_df = pd.read_csv(**kwargs)
+        dataset_df_lst = dataset_df.values.tolist()
+        self.dataset = gluon.data.SimpleDataset(dataset_df_lst)
+        split_ratio = kwargs.get('split_ratio', None) if 'split_ratio' in kwargs else 0.8
+        self.train_sampler, self.dev_sampler = get_split_samplers(self.dataset, split_ratio=split_ratio)
+
+    def _index_label_column(self, **kwargs):
+        label_column_id = kwargs.get('label', None)
+        if label_column_id is None:
+            return len(self.dataset[0]) - 1
```

Comment:
Either way works for me. I prefer the current implementation as normally the last column is the label column. I would incorporate this in the documentation and add warning messages to the users.

Reply:
This is a blocking feature for me which needs to be changed for my PR approval. We need the user to be able to specify the label column, it is not safe to say that the last column will always be the label column, and it will be very tedious for the user to reformat their data to work in that case.

Reply:
`usecols` is documented as an optional parameter, yet is required if the user passes str value label_column_id. This isn't necessary. You can get it directly from the DataFrame itself by df.columns().

Regarding your comment: More flexible does not equate to more user-friendly. I think we should never infer label position, it should be a required parameter, as it is in all other tasks in AutoGluon and in virtually all well known ML algorithms such as XGBoost, LightGBM, CatBoost, sklearn, Keras, etc.